module-bsp/board/rt1051/bsp/magnetometer/magnetometer.cpp

```diff
@@ -14,102 +14,106 @@
+        };
+
+        union i2c_buf_t
+        {
+            std::uint8_t buf[sizeof(als31300::whole_reg_t)];
+            als31300::whole_reg_t whole_reg;
+        };
 
-            TimerHandle_t timerHandle;
-            constexpr uint16_t magnetometerPollIntervalMs = 500;
+        constexpr std::uint16_t magnetometerPollIntervalMs = 500;
 
-            void TimerHandler(TimerHandle_t xTimer)
-            {
-                if (qHandleIrq != nullptr) {
-                    uint8_t val = 0x01;
-                    xQueueSend(qHandleIrq, &val, 0);
-                }
+        I2CAddress i2cAddress = {.deviceAddress = als31300::I2C_ADDRESS, .subAddress = 0, .subAddressSize = 1};
+        std::shared_ptr<drivers::DriverI2C> i2c;
+        i2c_buf_t i2cBuffer;
+
+        xQueueHandle qHandleIrq = nullptr;
+        TimerHandle_t pollingTimer;
+
+        bsp::KeyCodes currentParsedKey   = bsp::KeyCodes::Undefined;
+        bsp::KeyCodes lastSliderPosition = bsp::KeyCodes::Undefined;
+        Measurements lastMeasurement{};
+
+        bool isTimeToCompleteWriteDefinedForRegistry(std::uint8_t address)
+        {
+            const auto it = std::find(als31300::EEPROM_REGS.begin(), als31300::EEPROM_REGS.end(), address);
+            return it != als31300::EEPROM_REGS.end();
+        }
+
+        void timerHandler([[maybe_unused]] TimerHandle_t xTimer)
+        {
+            if (qHandleIrq != nullptr) {
+                std::uint8_t val = 0x01;
+                xQueueSend(qHandleIrq, &val, 0);
             }
-        } // namespace
+        }
 
-        bool i2cRead(const uint8_t reg_addr, als31300::whole_reg_t &whole_reg)
+        bool i2cRead(std::uint8_t regAddr, als31300::whole_reg_t &wholeReg)
         {
-            addr.subAddress = reg_addr;
-            if (i2c->Read(addr, i2c_buf.buf, sizeof(als31300::whole_reg_t)) != sizeof(als31300::whole_reg_t)) {
+            if (i2c == nullptr) {
+                return false;
+            }
+
+            i2cAddress.subAddress = regAddr;
+            if (i2c->Read(i2cAddress, i2cBuffer.buf, sizeof(als31300::whole_reg_t)) != sizeof(als31300::whole_reg_t)) {
                 return false;
             }
 #if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
-            // magnetometr talks big endian
-            i2c_buf.whole_reg = swapBytes(i2c_buf.whole_reg);
+            // Magnetometer talks big endian
+            i2cBuffer.whole_reg = utils::swapBytes(i2cBuffer.whole_reg);
 #endif
-            whole_reg = i2c_buf.whole_reg;
+            wholeReg = i2cBuffer.whole_reg;
             return true;
         }
 
-        bool i2cWrite(const uint8_t reg_addr, const als31300::whole_reg_t whole_reg)
+        bool i2cWrite(std::uint8_t regAddr, const als31300::whole_reg_t wholeReg)
         {
-            addr.subAddress = reg_addr;
+            if (i2c == nullptr) {
+                return false;
+            }
+
+            i2cAddress.subAddress = regAddr;
 #if __BYTE_ORDER__ != __ORDER_BIG_ENDIAN__
```

Comment:
This snippet is used twice - maybe make it a small function?

Reply:
You mean the one with `nullptr` check and setting subaddress?

Reply:
Done